### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by olpapina**

### CODE REVIEW

**Comment by TatsianaLohinava** [comment]
- I'd suggest to omit FindBy for the whole class, if you're not using it
- To avoid code repetition, it's better to put waiteIsClickable method in an abstract class and extend the classes that use it from this one, also probably use this method in other methods instead of using new wait strategy each time. Note a possible type in method name(?)

**Comment by olpapina** [comment]
> 

resolved code according to Tatsiana comments

File: src/main/java/com/solvd/eshop/page/BrandProductPage.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BrandProductPage extends AbstractPage {
+
+    @FindBy(css = ".result__link j-ga_track .result__name")
+    List<WebElement> brandProducts;
+
+    @FindBy(xpath = "//*[contains(@class,'cr-result__full')]//*[contains(@class,'result__price')]//*[contains(@class,'j-item-data')]")
+    List<WebElement> productPrices;
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/BrandProductPage.java]: Don't forget about access modifiers

File: src/main/java/com/solvd/eshop/page/HomePage.java [hunk 2]
```diff
@@ -3,50 +3,55 @@
...
-            waiter.until(ExpectedConditions.elementToBeClickable(loginButton));
-            if (loginButton.isDisplayed()) {
-                loginButton.click();
-            }
-            return loginPage;
+    public void clickAccountMenuAfterLogin() {
+        if (accountMenuAfterLogin.isDisplayed()) {
+            elementClick(accountMenuAfterLogin, 5);
         }
+    }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/HomePage.java]: If you add **if (someElement.isDisplayed())** before click on this element, it'll be difficult to understand why test failed. For me better variant to move this isDisplayed to **elementClick** method and add some exception or at least logger with message that element isn't visible.

File: src/main/java/com/solvd/eshop/page/LoginPage.java [hunk 3]
```diff
@@ -3,30 +3,44 @@
...
+    }
+
+    public void typeLoginPassword(String password) {
+        typeText(passwordField,password, 5);
+    }
+
+    public HomePage clickEnterButton() {
+        if (enterButton.isDisplayed()) {
+            elementClick(enterButton, 5);
+        }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/LoginPage.java]: The same as previous comment

File: src/main/java/com/solvd/eshop/page/ProductPage.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    List<WebElement> checkboxes;
+
+    @FindBy(css = "#j-filter__btn")
+    WebElement showProductsButton;
+
+    @FindBy(xpath = "//*[contains(@class,'cr-filter__price')]//*[@class='g-form__inputwrap']//*[@name='filter[price][from]']")
+    WebElement minPriceField;
+
+    @FindBy(xpath = "//*[contains(@class,'cr-filter__price')]//*[@class='g-form__inputwrap']//*[@name='filter[price][to]']")
+    WebElement maxPriceField;
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/ProductPage.java]: Don't forget about access modifiers

File: src/main/java/com/solvd/eshop/page/RegistrationPrivacyPage.java [hunk 5]
```diff
@@ -3,30 +3,20 @@
...
+        super(driver);
     }
 
     public SuccessRegistrationPage clickPrivacyButton() {
         SuccessRegistrationPage successRegistrationPage = new SuccessRegistrationPage(driver);
-        waiter.until(ExpectedConditions.elementToBeClickable(privacyButton));
         if (privacyButton.isDisplayed()) {
-            privacyButton.click();
+            elementClick(privacyButton, 5);
         }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/RegistrationPrivacyPage.java]: Please have a look at above comment related to isDisplayed and click

File: src/main/java/com/solvd/eshop/page/ResultPage.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ResultPage extends AbstractPage {
+    @FindBy(css = ".result__name")
+    List<WebElement> searchResultItems;
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/ResultPage.java]: Don't forget about access modifiers

**Review by marinagavrilenko — COMMENTED** [review]
Not a lot of changes are needed.

File: src/main/java/com/solvd/eshop/page/BrandProductPage.java (hunk 1 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/page/BrandProductPage.java]: resolved

File: src/main/java/com/solvd/eshop/page/HomePage.java (hunk 2 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/page/HomePage.java]: someElement.isDisplayed()) was moved to elementClick method and logger was added

File: src/main/java/com/solvd/eshop/page/ResultPage.java (hunk 6 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/page/ResultPage.java]: resolved

File: src/main/java/com/solvd/eshop/page/ProductPage.java (hunk 4 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/page/ProductPage.java]: resolved

File: src/main/java/com/solvd/eshop/elements/SearchSection.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void typeTextInSearch(String searchText) {
+
+        typeText(searchField, searchText, 5);
+    }
+
+    public ResultPage clickSearchButton() {
+        if (searchButton.isDisplayed()) {
+            elementClick(searchButton, 5);
+        }
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/elements/SearchSection.java]: This element click contains if, you can remove it

File: src/main/java/com/solvd/eshop/page/AbstractPage.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public void elementClick(WebElement webElement, long second) {
+        new WebDriverWait(this.driver, Duration.ofSeconds(second))
+                .until((ExpectedConditions.elementToBeClickable(webElement)));
+        if (webElement.isDisplayed()) {
+            webElement.click();
+            LOGGER.info(webElement.getTagName() + " is clicked");
+        } else {
+            LOGGER.info(webElement.getTagName() + " isn't clicked");
```
Comment (marinagavrilenko) [inline src/main/java/com/solvd/eshop/page/AbstractPage.java]: Better use LOGGER.error

File: src/main/java/com/solvd/eshop/elements/SearchSection.java (hunk 7 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/elements/SearchSection.java]: resolved

File: src/main/java/com/solvd/eshop/page/AbstractPage.java (hunk 8 above)
Comment (olpapina) [reply src/main/java/com/solvd/eshop/page/AbstractPage.java]: resolved

**Review by marinagavrilenko — APPROVED** [review]
You can merge this PR :)

**Comment by olpapina** [comment]
> You can merge this PR :)

Thank you)

**Pull request merged**